enhancements/baremetal/enable-baremetal-on-other-platforms.md

```diff
@@ -0,0 +1,197 @@
+in edge topologies, which without this feature, is restricted to having the
+central OpenShift cluster deployed on baremetal.
+
+See:
+- https://github.com/openshift/enhancements/blob/master/enhancements/installer/agent-based-installation-in-hive.md
+- https://github.com/openshift/assisted-service/tree/master/docs/hive-integration
+
+### Goals
+
+The specific goals of this proposal are to:
+
+Support the centralized host management use case by partially enabling Baremetal Host API
+on the following platforms:
+- None
+- OpenStack
+- vSphere
+
+We will be successful when:
+
+Centralized host management can deploy clusters when running on the above platforms.
+
+### Non-Goals
+
+Allow Baremetal Host API to be fully enabled on all platforms.
+
+Note that the scope of this epic is only enablement of the Metal3
+BaremetalHost API and associated controller/services - the centralized host management flow
+currently interacts directly via this API without any Machine API integration.
+
+## Proposal
+
+BMO (baremetal-operator) provides the Baremetal Host API, it in turn is configured
+and managed by CBO (cluster-baremetal-operator).
+
+CBO reads the Provisioning CR that is created by the installer on baremetal platforms
+and uses that to configure and deploy BMO. In the case of non-baremetal platforms
+the user (or automation) will need to define the Provisioning CR.
+
+Currently CBO checks the platform and if it is not baremetal it will be in a "disabled" state i.e. it will
+1. set status.conditions Disabled=true and
+2. not read or process the Provisioning CR and thus not deploy baremetal-operator.
+
+This proposal is to allow CBO to be enabled on the following platforms only:
+- Baremetal (current)
+- None
+- OpenStack
+- vSphere.
+
+Note: The Hypershift use case will be explicitly disallowed by disabling CBO when
+Infrastructure.Status.ControlPlaneTopology == "External".
+
+Further (to restrict the testing matrix) the allowed configuration options
+of the Provisioning CR will be restricted to exactly those required by centralized host management.
+
+*Only spec.provisioningNetwork=Disabled mode will be accepted in the Provisioning CR.*
+
+If any other provisioningNetwork mode is set, the CBO webhook will refuse the change
+in the usual case, but if defined before upgrading the operator, the Reconcile loop
+must always validate the Provisioning CR.
+
+Note:
+
+1. when the Provisioning CR is set to provisioningNetwork=Disabled mode, worker
+nodes would be booted via virtual media. This removes the requirement for the
+Provisioning Network which can be expected to be available only in Baremetal platform types.
+
+2. documentation will need to be added to the centralized host management documentation
+explaining how to create a Provisioning CR. Current documentation is here:
+https://github.com/openshift/assisted-service/blob/8880093ef5ce041d4c1951ffd5ea1096991ec3ee/docs/user-guide/assisted-service-on-openshift.md#configure-bare-metal-operator
```

Comment:
This link updates an existing Provisioning CR to start watching all Namespaces. When running on Platforms, OpenStack, vSphere and None, a completely new Provisioning CR needs to be created. Documentation needs to be updated with information from https://github.com/openshift/cluster-baremetal-operator/blob/master/README.md to define a valid Provisioning CR for these platforms.

Reply:
On these platforms, will CBO be able to generate a Provisioning CR with some valid defaults so that it can be updated later by the assisted-service?  This could be handled as an implementation detail that can be worked out later.